### PR TITLE
Changes from parity testing

### PR DESCRIPTION
**Pull request by ancasarb**
These changes (the ones around OneHotEncoder) are after I've tested roughly 50 models serialized using MLeap 0.10.3 for parity (especially around OneHotEncoder). Attempts to fix https://github.com/combust/mleap/issues/667 as well. 

Main changes:
- make all transformers that support handleInvalid use the trait HandleInvalid (including the ones that only support only keep and error)
- ensure transformers like Bucketizer(s) support skip correctly, by skipping the row in the result
- attempt again to fix parity of old/deprecated in 2.4.x OneHotEncoder between Spark, Mleap and Scikit-learn
- documentation updates (removing also outdated xgboost documentation)
- updating the release notes

Note: to be merged after https://github.com/combust/mleap/pull/673 is reviewed and merged, so that VectorIndexer can use HandleInvalid as well.

### CODE REVIEW

**Comment by ancasarb** [comment]
@WeichenXu123 @lucagiovagnoli I will create a spark 3.0.0 branch, so that @WeichenXu123 can start raising PRs and merging changes into that branch.

I will do this first thing tomorrow, after merging these three changes:
- https://github.com/combust/mleap/pull/673
- https://github.com/combust/mleap/pull/677
- https://github.com/combust/mleap/pull/645

Does this sound like a good plan? Please let me know if you have any worries or concerns. Thank you both for all your help!

File: mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala [hunk 1]
```diff
@@ -30,18 +30,19 @@ class OneHotEncoderOp extends MleapOp[OneHotEncoder, OneHotEncoderModel] {
 
     override def load(model: Model)
                      (implicit context: BundleContext[MleapContext]): OneHotEncoderModel = {
+      val dropLast = model.value("drop_last").getBoolean
       if (model.getValue("size").nonEmpty) {
         // Old version of 1HE.
+        val handleInvalid = if (dropLast) HandleInvalid.Keep else HandleInvalid.default
```
Comment (voganrc) [inline mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala:36]: You hard-code `attributes.append(('drop_last', True))` in `data.py`, so this removes the choice between `HandleInvalid.Keep` and `HandleInvalid.Error`.

Instead, we should just read whatever the bundle has inside of it, and have the Python class serialize:
```
if transformer.handle_unknown == 'ignore':
    attributes.append(('handle_invalid', 'keep'))
else:
    attributes.append(('handle_invalid', 'error'))
attributes.append(('drop_last', True))
```

File: mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala [hunk 2]
```diff
@@ -30,18 +30,19 @@ class OneHotEncoderOp extends MleapOp[OneHotEncoder, OneHotEncoderModel] {
...
         OneHotEncoderModel(categorySizes = Array(model.value("size").getLong.toInt),
-          // assume HandleInvalid.Keep always since old version of 1HE didn't have a way to specify this
-          handleInvalid = HandleInvalid.Keep,
-          dropLast = model.value("drop_last").getBoolean)
+          dropLast = dropLast,
+          handleInvalid = handleInvalid)
```
Comment (voganrc) [inline mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala:39]: nit: order of arguments is different from the `else` branch. Also you create a `handleInvalid` variable here but inline it in the assignment below

Comment (ancasarb) [reply mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala]: Fixed.

File: mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala (hunk 1 above)
Comment (ancasarb) [reply mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala]: You're right, I've amended the code.

**Comment by ancasarb** [comment]
@voganrc @lucagiovagnoli thank you for your review, I've addressed the comments, could you please take another look? thanks again!

File: mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala [hunk 3]
```diff
@@ -30,18 +30,22 @@ class OneHotEncoderOp extends MleapOp[OneHotEncoder, OneHotEncoderModel] {
...
-        OneHotEncoderModel(categorySizes = Array(model.value("size").getLong.toInt),
-          // assume HandleInvalid.Keep always since old version of 1HE didn't have a way to specify this
-          handleInvalid = HandleInvalid.Keep,
-          dropLast = model.value("drop_last").getBoolean)
+        OneHotEncoderModel(
+          categorySizes = Array(model.value("size").getLong.toInt),
+          handleInvalid =
+            model.getValue("handle_invalid").map(_.getString).map(HandleInvalid.fromString(_, false))
+              //this maintains backwards compatibility for models that don't have handle_invalid serialized
+              .getOrElse(if (dropLast) HandleInvalid.Keep else HandleInvalid.default),
```
Comment (voganrc) [inline mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala:41]: Should we make this `.getOrElse(HandleInvalid.Keep)` to match the hard-coded `handleInvalid = HandleInvalid.Keep` from before? 

Let's say there's an old bundle that had `drop_last = False` in it. If they were to run that bundle through the old code they would get an extra column for invalid data. But with the new code the same bundle would error when it sees invalid data.

Comment (ancasarb) [reply mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala:41]: Unfortunately, HandleInvalid.error needs to be the default for old spark models to be backwards compatible. I made previously the change to just use HandleInvalid.Keep thinking it works for Spark models too, but it doesn't :( It's been a bit of a nightmare to get this 1he updates working.

**Pull request merged**